sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -3098,4 +3098,20 @@ class HiveDDLSuite
       "CREATE TABLE tab (c1 int) PARTITIONED BY (c1) STORED AS PARQUET",
       "Cannot use all columns for partition columns")
   }
+
+  test("SPARK-43359: Delete table not allowed") {
+    val tbl = "T1"
+    withTable(tbl) {
+      sql(s"CREATE TABLE $tbl(c1 INT)")
+      val e = intercept[AnalysisException] {
+        sql(s"DELETE FROM $tbl WHERE c1 = 1")
+      }
+      checkError(e,
+        errorClass = "UNSUPPORTED_FEATURE.TABLE_OPERATION",
+        parameters = Map(
+          "tableName" -> "`spark_catalog`.`default`.`t1`",
```

Comment:
How about this one?

Reply:
Ok, chang to s"`spark_catalog`.`default`.`$tbl`"